### DISCUSSION TITLE
Cannot find module '@smithy/protocol-http'\

### DISCUSSION THREAD

**Discussion by noumanbhatti** [aws/aws-sdk-js-v3#5474] [Q&A]
I've updated the "@aws-sdk/client-chime-sdk-meetings": "3.445.0", and now I'm getting the following error. Even if I lower down the version to a year ago issue is same 

{
    "errorType": "Runtime.ImportModuleError",
    "errorMessage": "Error: Cannot find module '@smithy/protocol-http'\nRequire stack:\n- /var/task/node_modules/@aws-sdk/middleware-host-header/dist-cjs/index.js\n- /var/task/node_modules/@aws-sdk/client-chime-sdk-meetings/dist-cjs/ChimeSDKMeetingsClient.js\n- /var/task/node_modules/@aws-sdk/client-chime-sdk-meetings/dist-cjs/index.js\n- /var/task/callscreen/createMeeting.js\n- /var/runtime/index.mjs",
    "stack": [
        "Runtime.ImportModuleError: Error: Cannot find module '@smithy/protocol-http'",
        "Require stack:",
        "- /var/task/node_modules/@aws-sdk/middleware-host-header/dist-cjs/index.js",
        "- /var/task/node_modules/@aws-sdk/client-chime-sdk-meetings/dist-cjs/ChimeSDKMeetingsClient.js",
        "- /var/task/node_modules/@aws-sdk/client-chime-sdk-meetings/dist-cjs/index.js",
        "- /var/task/callscreen/createMeeting.js",
        "- /var/runtime/index.mjs",
        "    at _loadUserApp (file:///var/runtime/index.mjs:1087:17)",
        "    at async Object.UserFunction.js.module.exports.load (file:///var/runtime/index.mjs:1119:21)",
        "    at async start (file:///var/runtime/index.mjs:1282:23)",
        "    at async file:///var/runtime/index.mjs:1288:1"
    ]
}

**Comment by aBurmeseDev** [comment]
Hi @noumanbhatti - sorry for not getting to you sooner. Are you still running into this error? If so, can you share you code so that we can further investigate?

**Reply by noumanbhatti** [reply]
na i think I've resolved the issue

**Reply by simonbutn** [reply]
Hi there @noumanbhatti . How were you able to resolve the issue?

**Reply by MichaelTrosin** [reply]
It seems that this is also meant to run on AWS Lambda. In that case, I was able to resolve this by putting the package that the error message is complaining about into package.json, and then under `NodeJsFunction({ bundling: { nodeModules: [ 'here-as-well'] }})`. Deploy it and then each time it gives you a different package that needs to be added to both files until it's all resolved.

My final `nodeModules` array looked like this:
```
nodeModules: ['@smithy/util-body-length-node', '@smithy/smithy-client', '@aws-smithy/server-common', '@smithy/protocol-http', "re2-wasm"]
```

(though I'm writing a Smithy server so the final list may vary for you)
